troubleshooting-and-how-tos/deployment.md

```diff
@@ -0,0 +1,50 @@
+# Recipe for One-Click Deployment
+
+This document contains a brief explanation on how we have setup one-click deployment. By one-click deployment, we mean by just making a release on Github, the changes are deployed without any human action. Many of the items used from the AWS menu are explained below.
+
+- **<u>Docker</u>**: 
+
+  Both the frontend and backend are containerized in a node-14 base image, with a Dockerfile akin to this [one](https://github.com/rakaar/dockerize-react-and-express-app/blob/master/Dockerfile).
+
+- **<u>Github Actions</u>**: 
+
+  On release, a Github Action is triggered. The YAML file action has been inspired from the [C-3PO project from LTTKGP](https://github.com/lttkgp/C-3PO/blob/master/.github/workflows/deploy.yml). This Actions in brief do the following (The AWS terms will be explained soon)
+
+  - Logging into AWS using the secrets configured via Github Secrets
+  - Builds and pushes the image to **Elastic Container Registry(ECR)**
+  - Deploy the image via **Elastic Container Service(ECS) Fargate**
+
+- <u>**Elastic Container Registry(ECR)**</u>: 
+
+  AWS Service where you can store Docker Container Images. The Docker images are stored in a private repository. To look at the images in AWS, visit ECR by searching in the console. Click on the repository name. 
+
+- **<u>Elastic Container Service(ECS) Fargate</u>**: 
+
+  Once the latest image is stored in the ECR, they need to be deployed. A service by name **ECS Fargate** handles this job of deploying the docker image. ECS Fargate is configured using a JSON file present in the `deploy` folder. The JSON is obtained from the AWS Console(Instructions to obtain JSON from AWS console are pinned in Felvin Slack). 
```

Comment:
Consider all slack conversation transient. It will go away. Please add another doc for these instructions.

Reply:
Added the doc and referenced it!